pkg/prometheus/promcfg.go

```diff
@@ -534,14 +534,14 @@ func (cg *configGenerator) generatePodMonitorConfig(
 		case metav1.LabelSelectorOpExists:
 			relabelings = append(relabelings, yaml.MapSlice{
 				{Key: "action", Value: "keep"},
-				{Key: "source_labels", Value: []string{"__meta_kubernetes_pod_label_" + sanitizeLabelName(exp.Key)}},
-				{Key: "regex", Value: ".+"},
+				{Key: "source_labels", Value: []string{"__meta_kubernetes_pod_labelpresent_" + sanitizeLabelName(exp.Key)}},
```

Comment:
With this change, any actual values of labels would be lost. I think we need a combined `regex` something along the lines of:

```
{Key: "source_labels", Value: []string{"__meta_kubernetes_pod_labelpresent_" + sanitizeLabelName(exp.Key), "__meta_kubernetes_pod_label_" + sanitizeLabelName(exp.Key)}},
{Key: "regex", Value: "true;.+"},
```

Reply:
Thanks! I've updated the code accordingly.

Reply:
I was slightly off the first time around actually, sorry for not realizing this the first time around. It should actually be

{Key: "source_labels", Value: []string{"__meta_kubernetes_pod_labelpresent_" + sanitizeLabelName(exp.Key)}},
{Key: "regex", Value: "true"},

Reply:
No worries, I didn't bother myself to check documentation on this topic, so thanks for taking care of it. I'll change the code.